pymeasure/adapters/telnet.py

```diff
@@ -1,109 +0,0 @@
-#
-# This file is part of the PyMeasure package.
-#
-# Copyright (c) 2013-2023 PyMeasure Developers
-#
-# Permission is hereby granted, free of charge, to any person obtaining a copy
-# of this software and associated documentation files (the "Software"), to deal
-# in the Software without restriction, including without limitation the rights
-# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-# copies of the Software, and to permit persons to whom the Software is
-# furnished to do so, subject to the following conditions:
-#
-# The above copyright notice and this permission notice shall be included in
-# all copies or substantial portions of the Software.
-#
-# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-# THE SOFTWARE.
-#
-
-import telnetlib
-import time
-from warnings import warn
-
-from .adapter import Adapter
-
-
-class TelnetAdapter(Adapter):
```

Comment:
@bmoneke can we replace the `TelnetAdapter` implementation with a deprecation notice and an equivalently functioning/configured `VISAAdapter` subclass? That way people that currently use the `TelnetAdapter` (outside of the ANC300) would not get broken code immediately?

Reply:
I meant transitioning the ANC300 to the "new way" as we are doing here, but in parallel deprecate (not just remove) `TelnetAdapter` somewhat nicely for users that might be using it elsewhere currently, by replacing it with an equivalent implementation using `VisaAdapter` (+ a notice).

Reply:
If that's not feasible, then :shrug: we can just remove it (but it would be nice to leave a better hint than an `AttributeError` or an `ImportError` - maybe a barebones class that raises a FutureWarning with info when used?

Reply:
This is a good idea, I might have jumped the gun on just chucking the whole thing. I think reworking the whole thing so it can become deprecated is maybe not valuable. Maybe add a deprecation message indicating to use the VISAAdapter and maybe something more informative maybe linking to the help I was provided here earlier or some similar article.

Reply:
For the ANC300, we can add a "host" kwarg (and make the adapter kwarg optional). If adapter is None, a deprecation warning will be raised (see #672 on how to implement it) and the host kwarg will be used to create the resource string.

> If that's not feasible, then 🤷 we can just remove it (but it would be nice to leave a better hint than an AttributeError or an ImportError - maybe a barebones class that raises a FutureWarning with info when used?

I don't see a problem keeping the TelnetAdapter in the codebase for some time. Therefore, the TelnetAdapter itself should stay in:  at least with a deprecation warning, better as a subclass of VISAAdapter. I don't know, though, whether the rework of the TelnetAdapter (without testing possibilities) is work the effort.

Reply:
> For the ANC300, we can add a "host" kwarg (and make the adapter kwarg optional). If adapter is None, a deprecation warning will be raised (see #672 on how to implement it) and the host kwarg will be used to create the resource string.

This is done.

> as a subclass of VISAAdapter.
Calling a connection `telnet (socket)` that is actually `VISA` is misleading and could cause confusion about the arguments available and how transmission terminations are handled. It's probably better to leave the TelnetAdapter how it was and remove it before python 3.13 rolls around. If it is later desired to keep it for any reason, switch to some implementation of `SocketAdapter` using sockets.

Reply:
Added this adapter back. Let me know if this is sufficient to resolve.